producers/weather-replayer/src/main/java/org/bptlab/cepta/producers/weatherreplayer/WeatherDataReplayerServer.java

```diff
@@ -0,0 +1,192 @@
+import org.bptlab.cepta.schemas.grpc.GrpcServer;
+import org.bptlab.cepta.serialization.AvroBinarySerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings({"OptionalUsedAsFieldOrParameterType", "FieldCanBeLocal"})
+
+public class WeatherDataReplayerServer
+    extends GrpcServer<org.bptlab.cepta.producers.replayer.ReplayerGrpc.ReplayerImplBase> {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(WeatherDataReplayerServer.class.getName());
+
+  public static final class Builder {
+
+    private int servicePort;
+    private KafkaConfig kafkaConfig = new KafkaConfig();
+    private PostgresConfig databaseConfig = new PostgresConfig();
+    private long frequency = 5000;
+    private Optional<String> mustMatch;
+    private Optional<Timestamp> startTimestamp;
+    private Optional<Timestamp> endTimestamp;
+
+    private Builder() {
+    }
+
+    public Builder withDatabaseConfig(PostgresConfig config) {
+      databaseConfig = config;
+      return this;
+    }
+
+    public Builder withKafkaConfig(KafkaConfig config) {
+      kafkaConfig = config.withKeySerializer(Optional.of(LongSerializer::new)).withValueSerializer(Optional.of(
+          AvroBinarySerializer<WeatherData>::new));
+      return this;
+    }
+
+    public Builder withStartTime(Optional<Timestamp> startTime) {
+      startTimestamp = startTime;
+      return this;
+    }
+
+    public Builder withEndTime(Optional<Timestamp> endTime) {
+      endTimestamp = endTime;
+      return this;
+    }
+
+    public Builder withFrequency(long frequency) {
+      this.frequency = frequency;
+      return this;
+    }
+
+    public Builder mustMatch(Optional<String> condition) {
+      this.mustMatch = condition;
+      return this;
+    }
+
+    public WeatherDataReplayerServer build(int servicePort) {
+      WeatherDataReplayer WeatherDataReplayer = new WeatherDataReplayer(
+          kafkaConfig.withClientId("WeatherDataReplayerClient").getProperties(),
+          Topics.WEATHER_DATA);
+
+      PostgresReplayer[] replayers = new PostgresReplayer[]{
+          WeatherDataReplayer
+      };
+
+      for (PostgresReplayer replayer : replayers) {
+        replayer.connect(databaseConfig);
+        replayer.setFrequency(frequency);
+        replayer.setMustMatch(this.mustMatch);
+        startTimestamp.ifPresent(replayer::setStartTime);
+        endTimestamp.ifPresent(replayer::setEndTime);
+      }
+      return new WeatherDataReplayerServer(replayers, servicePort);
+    }
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public WeatherDataReplayerServer(PostgresReplayer replayer, int servicePort) {
+    super(new WeatherDataReplayerService(new PostgresReplayer[]{replayer}), servicePort);
+  }
+
+  public WeatherDataReplayerServer(PostgresReplayer[] replayers, int servicePort) {
+    super(new WeatherDataReplayerService(replayers), servicePort);
+  }
+
+  private static class WeatherDataReplayerService
```

Comment:
We should maybe add different grpc schemas for each replayer for the sake of future proofing and only keed a minimum of interfaces shared